renku/service/views/decorators.py

```diff
@@ -193,11 +198,18 @@ def decorated_function(*args, **kwargs):
 
             error_code = GIT_ACCESS_DENIED_ERROR_CODE if "Access denied" in e.stderr else GIT_UNKNOWN_ERROR_CODE
 
-            # strip oauth tokens
-            error_reason = format(" ".join(e.stderr.strip().split("\n")))
-            error_reason_safe = re.sub("^(.+)(oauth2:)(.+)(@)(.+)$", r"\1\2<token-hidden>\4\5", error_reason)
+            if "is this a git repository?" in e.stderr:
+                error_reason = "Repository could not be found"
+            elif "Access denied" in e.stderr:
+                error_reason = "Repository could not be accessed - Do you have access rights?"
+            else:
+                error_reason = format(" ".join(e.stderr.strip().split("\n")))
 
-            return jsonify(error={"code": error_code, "reason": f"git error: {error_reason_safe}"})
+                # strip oauth tokens
+                error_reason_safe = re.sub("^(.+)(oauth2:)(.+)(@)(.+)$", r"\1\2<token-hidden>\4\5", error_reason)
```

Comment:
```suggestion
                error_reason_safe = re.sub("^(.+oauth2:)[^@]+(@.+)$", r"\1<token-hidden>\2", error_reason)
```
To make the greedy regex not match more than it should in case there's multiple `@`'s.